src/pyk/ktool/kprint.py

```diff
@@ -362,10 +366,16 @@ def unparser_for_production(prod: KProduction) -> Callable[..., str]:
     def _unparser(*args: Any) -> str:
         index = 0
         result = []
+        num_nonterm = len([item for item in prod.items if type(item) is KNonTerminal])
+        num_named_nonterm = len([item for item in prod.items if type(item) is KNonTerminal and item.name != None])
         for item in prod.items:
             if type(item) is KTerminal:
                 result.append(item.value)
             elif type(item) is KNonTerminal and index < len(args):
+                if index == 0 and num_nonterm == num_named_nonterm and num_nonterm > 0:
+                    result.append('...')
+                if num_nonterm == num_named_nonterm and num_nonterm > 0:
+                    result.append(f'{item.name}:')
```

Comment:
I guess we don't need to check `num_nonterm > 0` here, because we're inside an iteration of the loop where `type(item) is KNonTerminal`.

```suggestion
                if index == 0 and num_nonterm == num_named_nonterm:
                    result.append('...')
                if num_nonterm == num_named_nonterm:
                    result.append(f'{item.name}:')
```